framework/reporter.go

```diff
@@ -0,0 +1,278 @@
+/*
+ Copyright 2024 The OSCAL Compass Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package framework
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/defenseunicorns/go-oscal/src/pkg/uuid"
+	oscalTypes "github.com/defenseunicorns/go-oscal/src/types/oscal-1-1-2"
+	"github.com/hashicorp/go-hclog"
+	"github.com/oscal-compass/compliance-to-policy-go/v2/framework/config"
+	"github.com/oscal-compass/compliance-to-policy-go/v2/policy"
+	"github.com/oscal-compass/oscal-sdk-go/extensions"
+	"github.com/oscal-compass/oscal-sdk-go/models"
+	"github.com/oscal-compass/oscal-sdk-go/rules"
+	"github.com/oscal-compass/oscal-sdk-go/settings"
+)
+
+const (
+	defaultVersion = "0.1.0"
+	defaultTitle   = "Automated Assessment Results"
+)
+
+type Reporter struct {
+	log        hclog.Logger
+	rulesStore rules.Store
+}
+
+func NewReporter(cfg *config.C2PConfig) (*Reporter, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+
+	rulesStore, _, err := config.ResolveOptions(cfg)
```

Comment:
Something that we might want to look into later - using `sync.Once.Do` to avoid indexing the ruleStore multiple times.